src/GeolocatorService.Shared/GeolocatorService.cs

```
@@ -26,7 +26,72 @@ public async Task<bool> GetIsPermissionGranted(CancellationToken ct)
 		public async Task<Geocoordinate> GetLocation(CancellationToken ct)
 		{
 			var position = await _locator.GetGeopositionAsync().AsTask(ct);
-			return position.Coordinate;
+			var coordinate = GetGeoCoordinates(position.Coordinate);
+			return coordinate;
+		}
+
+		private Geocoordinate GetGeoCoordinates(Windows.Devices.Geolocation.Geocoordinate coordinate)
+		{
+			return new Geocoordinate(
+				accuracy: coordinate.Accuracy,
+				altitude: coordinate.Altitude,
+				altitudeAccuracy: coordinate.AltitudeAccuracy,
+				heading: coordinate.Heading,
+				latitude: coordinate.Latitude,
+				longitude: coordinate.Longitude,
+				speed: coordinate.Speed,
+				timeStamp: coordinate.Timestamp,
+				point: GetPoint(coordinate.Point),
+				positionSource: GetPositionSource(coordinate.PositionSource),
+				positionSourceTimestamp: coordinate.PositionSourceTimestamp);
+		}
+
+		private PositionSource GetPositionSource(Windows.Devices.Geolocation.PositionSource positionSource)
+		{
+			switch(positionSource)
+			{
+				case Windows.Devices.Geolocation.PositionSource.Cellular:
+					return PositionSource.Cellular;
+				case Windows.Devices.Geolocation.PositionSource.Satellite:
+					return PositionSource.Satellite;
+				case Windows.Devices.Geolocation.PositionSource.WiFi:
+					return PositionSource.WiFi;
+				case Windows.Devices.Geolocation.PositionSource.IPAddress:
+					return PositionSource.IPAddress;
+				case Windows.Devices.Geolocation.PositionSource.Unknown:
+					return PositionSource.Unknown;
+				case Windows.Devices.Geolocation.PositionSource.Default:
+					return PositionSource.Default;
+				default:
```

You shouldn't use default for the last value. You could use default to throw an unsupported exception :)

---

Very true! changed it!